cpp/onedal/accessor.hpp

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+#pragma once
+
+#include "onedal/table_builder.hpp"
+
+namespace dal {
+
+template <typename AccessType>
+class row_accessor {
+private:
+    using storage_t = detail::get_dense_storage_iface_t<AccessType>;
+
+public:
+    using data_t = std::remove_const_t<AccessType>;
+    static constexpr bool is_readonly = std::is_const_v<AccessType>;
+
+    template <typename T = AccessType>
+    row_accessor(const table& t, std::enable_if_t<sizeof(T) && is_readonly>* = nullptr)
```

Comment:
Why `sizeof(T)` is needed here?

Reply:
I'd declare the constructor this way:
```cpp
template <typename = std::enable_if_t<sizeof(T) && is_readonly>>
row_accessor(const table& t);
```

Does it work?

Reply:
Yes, it works. `sizeof(T)` was needed for some purpose, but I cannot remember it right now. Since all tests are passing without it, I will remove it from `enable_if_t` statement